server/src/domain/asset/asset.service.ts

```diff
@@ -91,6 +93,34 @@ export class AssetService {
     this.configCore = SystemConfigCore.create(configRepository);
   }
 
+  search(authUser: AuthUserDto, dto: AssetSearchDto) {
+    let checksum: Buffer | undefined = undefined;
+
+    if (dto.checksum) {
+      const encoding = dto.checksum.length === 28 ? 'base64' : 'hex';
+      checksum = Buffer.from(dto.checksum, encoding);
+    }
+
+    const enumToOrder = { [AssetOrder.ASC]: 'ASC', [AssetOrder.DESC]: 'DESC' } as const;
+    const order = dto.order ? enumToOrder[dto.order] : undefined;
```

Comment:
I have a personal preference of being explicit rather than implicit in these situations. I think better transparency leads to less confusion/bugs. Plus, this uses typing, while string manipulation probably doesn't.